cluster_agent/identity/slurmrestd.py

```diff
@@ -76,21 +72,24 @@ def acquire_token(username: str) -> str:
     token = _load_token_from_cache(username)
 
     if token is None:
-        logger.debug("Attempting to acquire token from Slurmrestd")
-        proc = subprocess.Popen(
-            f"scontrol token username={username}".split(),
-            stdin=subprocess.PIPE,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-        )
-        stdout, stderr = proc.communicate()
-        ProcessExecutionError.require_condition(
-            proc.returncode == 0, stderr.decode().strip()
-        )
-        token = stdout.decode().strip().split("=")[1]
+        logger.debug("Attempting to generate token for Slurmrestd")
+        if SETTINGS.SLURMRESTD_USE_KEY_PATH:
+            secret_key = open(SETTINGS.SLURMRESTD_JWT_KEY_PATH, "r").read()
+        else:
+            secret_key = SETTINGS.SLURMRESTD_JWT_KEY_STRING
+
+        now = datetime.now()
```

Comment:
Need to be careful here. By default with `datetime` you will get a timestamp in your local timezone. It might be a better idea to use UTC timestamps in all cases to eliminate ambiguity.

Reply:
hmm, I don't think the timezone here has any practical effect since the agent runs in customers' clusters

Reply:
The issue I could see happening is if the cluster-agent runs in a different timezone than the slurm rest api. I just bring it up, because I have seen a lot of issues arise from the use of time zone naïve timestamps in python.

If you like, you can disregard this comment. Just know that if it ever ends up biting us, I will give you a bunch of hell about it! 😆 😆

Reply:
this is a goog point indeed. Let's discuss it.

So, is there any way to set the timezone which slurmrestd will use? @jamesbeedy do you have any take on this?

Reply:
I couldn't find any information about that on the [docs](https://slurm.schedmd.com/slurmrestd.html)